### PR TITLE
preserve file table for better error messages

### PR DESCRIPTION
**Pull request by ericsciple**

### CODE REVIEW

File: src/Runner.Listener/MessageListener.cs [hunk 1]
```diff
@@ -174,6 +174,7 @@ public async Task<TaskAgentMessage> GetNextMessageAsync(CancellationToken token)
                     if (message != null)
                     {
                         _lastMessageId = message.MessageId;
+                        Console.WriteLine(GitHub.Runner.Sdk.StringUtil.ConvertToJson(message));
```
Comment (ericsciple) [inline src/Runner.Listener/MessageListener.cs]: remove

File: src/Runner.Worker/StepsRunner.cs [hunk 2]
```diff
@@ -480,9 +478,10 @@ private void CompleteStep(IStep step, IStep nextStep, TaskResult? result = null,
 
         private PipelineTemplateEvaluator CreateTemplateEvaluator(IExecutionContext executionContext)
         {
+            Trace.Info($"Creating template evaluator. FileTable: {StringUtil.ConvertToJson(executionContext.FileTable)}");
```
Comment (ericsciple) [inline src/Runner.Worker/StepsRunner.cs]: remove this tracing

File: src/Sdk/DTPipelines/Pipelines/ObjectTemplating/PipelineTemplateEvaluator.cs [hunk 3]
```diff
@@ -324,6 +326,22 @@ private TemplateContext CreateContext(DictionaryContextData contextData)
                 TraceWriter = m_trace,
             };
 
+            // Add the file table
+            System.IO.File.AppendAllText(@"C:\temp\context.txt", $"\nAdding file table with {m_fileTable?.Count} entries");
```
Comment (ericsciple) [inline src/Sdk/DTPipelines/Pipelines/ObjectTemplating/PipelineTemplateEvaluator.cs]: remove this trace line

File: src/Sdk/DTPipelines/Pipelines/ObjectTemplating/PipelineTemplateEvaluator.cs [hunk 4]
```diff
@@ -324,6 +326,22 @@ private TemplateContext CreateContext(DictionaryContextData contextData)
...
+            if (m_fileTable?.Count > 0)
+            {
+                for (var i = 0 ; i < m_fileTable.Count ; i++)
+                {
+                    System.IO.File.AppendAllText(@"C:\temp\context.txt", $"\nAdding {m_fileTable[i]}");
```
Comment (ericsciple) [inline src/Sdk/DTPipelines/Pipelines/ObjectTemplating/PipelineTemplateEvaluator.cs]: remove this trace line

File: src/Sdk/DTPipelines/Pipelines/ObjectTemplating/PipelineTemplateEvaluator.cs [hunk 5]
```diff
@@ -324,6 +326,22 @@ private TemplateContext CreateContext(DictionaryContextData contextData)
...
+                    var fileId = result.GetFileId(m_fileTable[i]);
+                    if (fileId != i + 1)
```
Comment (ericsciple) [inline src/Sdk/DTPipelines/Pipelines/ObjectTemplating/PipelineTemplateEvaluator.cs]: delete this assertion

Comment (ericsciple) [reply src/Sdk/DTPipelines/Pipelines/ObjectTemplating/PipelineTemplateEvaluator.cs]: add unit test on server instead

File: src/Runner.Worker/ActionManifestManager.cs [hunk 6]
```diff
@@ -32,6 +32,7 @@ public interface IActionManifestManager : IRunnerService
     public sealed class ActionManifestManager : RunnerService, IActionManifestManager
```
Comment (ericsciple) [inline src/Runner.Worker/ActionManifestManager.cs]: test e2e for action manifest manager

File: src/Runner.Worker/ActionRunner.cs [hunk 7]
```diff
@@ -143,7 +143,7 @@ public async Task RunAsync()
             ExecutionContext.Debug("Loading inputs");
             var templateTrace = ExecutionContext.ToTemplateTraceWriter();
             var schema = new PipelineTemplateSchemaFactory().CreateSchema();
-            var templateEvaluator = new PipelineTemplateEvaluator(templateTrace, schema);
+            var templateEvaluator = new PipelineTemplateEvaluator(templateTrace, schema, ExecutionContext.FileTable);
```
Comment (ericsciple) [inline src/Runner.Worker/ActionRunner.cs]: consider move to extension method

**Comment by TingluoHuang** [comment]
need to fix L0... 😄

**Comment by ericsciple** [comment]
i tested back compat w/ old server

**Pull request merged**